pkg/api/types.go

```diff
@@ -1631,6 +1635,12 @@ type ObjectReference struct {
 	FieldPath string `json:"fieldPath,omitempty"`
 }
 
+// LocalObjectReference contains enough information to let you locate the referenced object inside the same namespace.
+type LocalObjectReference struct {
```

Comment:
I don't remember where discussion landed on this -- did we decide to omit 'Kind' for now and encode it in the field name?